library/Terminal42/ChangeLanguage/PageFinder.php

```diff
@@ -248,6 +276,27 @@ private function addPublishingConditions(array &$columns, $table)
     }
 
     /**
+     * @param array  $columns
+     * @param array  $values
+     * @param string $table
+     */
+    private function addLimitRootPageIdsCondition(array &$columns, array &$values, $table)
+    {
+        if (count($this->rootPageIds)) {
+            /**
+             * this will generate a comma separated string of placeholders
+             * example: if there are 5 entries in rootPageIds it will generate the following string: ?,?,?,?,?
+             */
+            $placeholders = (implode(',', array_pad([], count($this->rootPageIds), '?')));
+
+            $negation = ($this->negateRootPageIds)? 'NOT' : '';
+
+            $columns[] = "$table.id $negation IN($placeholders)";
```

Comment:
"$table.id ${negation} IN(${placeholders})"

should be this way imho

Reply:
This looked so weird to me I googled it. I found my answer and found this interesting tidbit aswell regarding performance of strings/concatenation : http://php.net/manual/de/language.types.string.php#120160

Reply:
see c3f098b